File: src_py/sender.py
# from src_py.aes_ops import encrypt_cbc
from src_py.aes_ops import encrypt_ecb

def main():
    key = b"thisisakey123456"

    message = input("Enter plaintext: ")
    plaintext = message.encode("utf-8")

   # iv = input("Enter iv:")
   # ciphertext, iv = encrypt_cbc(plaintext, key, iv=None)
    ciphertext = encrypt_ecb(key, plaintext)

    print("Private key:            ", key.hex())
    # print("IV (public):            ", iv.hex())
    print("Ciphertext (public):    ", ciphertext.hex())

if __name__ == "__main__":
    main()

File: src_mat/sender.m
key = uint8('thisisakey123456');  % 16-byte

message   = input('Enter plaintext: ', 's');
plaintext = uint8(message);

[ciphertext, iv] = encrypt_cbc(plaintext, key);

fprintf('Private key:          %s\n', bytes2hex(key));
fprintf('IV (public):          %s\n', bytes2hex(iv));
fprintf('Ciphertext (public):  %s\n', bytes2hex(ciphertext));

% ---- helper local function ----
function h = bytes2hex(b)
    b = uint8(b(:));
    h = upper(reshape(dec2hex(b), 1, []));   % mỗi byte -> 2 ký tự hex
end
